core-bundle/contao/forms/Form.php

```diff
@@ -319,7 +319,7 @@ protected function processFormData($arrSubmitted, $arrLabels, $arrFields, $arrFi
 			foreach ($GLOBALS['TL_HOOKS']['prepareFormData'] as $callback)
 			{
 				$this->import($callback[0]);
-				$this->{$callback[0]}->{$callback[1]}($arrSubmitted, $arrLabels, $arrFields, $this);
+				$this->{$callback[0]}->{$callback[1]}($arrSubmitted, $arrLabels, $arrFields, $this, $arrFiles);
```

Comment:
```suggestion
				$this->{$callback[0]}->{$callback[1]}($arrSubmitted, $arrLabels, $arrFields, $arrFiles, $this);
```

I know this is technically a BC break, however, we have changed all callbacks to pass `$this` as last argument in Contao 5 and therefore this could be seen as a bugfix.

Reply:
> I know this is technically a BC break, however, we have changed all callbacks to pass `$this` as last argument in Contao 5 and therefore this could be seen as a bugfix.

But in all these cases we only _added_ it at the end. I don't think we should break BC here as this provides no gain.

Reply:
I wouldn't do that either TBH. It's probably one of the most frequently used hooks so I'd just add it at the end.